### PR TITLE
test-infra: add presubmit file for prow job

### PR DESCRIPTION
**Pull request by nlandolfi**
This introduces a `/jobs` directory for scripts that are run as presubmit/postsubmit jobs on the prow continuous integration infrastructure. 

I have included a job that runs the steps of a pilot presubmit; namely, `jobs/pilot-presubmit.sh`. Loosely, this amounts to linting, building and _unit_ testing the code. 

Currently our Prow infrastructure listens to the pilot repository and schedules and runs the presubmit job, but does not report visibly (GitHub comment or PR status) the result. Once I am confident that the reporting is reasonable, and after clearing with leads, I can turn it on.

We intend to run Jenkins and Prow simultaneously, in the medium term, and to follow up, in other PRs, with the remaining jobs (integration tests, e2e test, code coverage). Once Prow has proven it fits the bill, we can consider winding down Jenkins. More details for the Prow migration [here](https://github.com/istio/test-infra/pull/253).

### CODE REVIEW

**Comment by kyessenov** [comment]
test failure is a flake, maybe run again?

**Comment by nlandolfi** [comment]
yeah, we decided on prow/ dir name instead of jobs, will update now

**Comment by nlandolfi** [comment]
update: most recent build log for prow [here](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_pilot/910/pilot-presubmit/21/). failing because of lack of permission to push to the gcr registry. 

@kyessenov does it make sense for prow to also build and push there?

**Comment by kyessenov** [comment]
`bin/e2e` pushes docker images to a registry so that the test cluster can use them.
I'm open to using a better solution, maybe a registry in the cluster or pushing images directly to the cluster, but letting the test script push debug images seems like the least amount of work.

File: prow/pilot-presubmit.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+set -ex
+
+# Test harness will checkout code to directory $GOPATH/src/github.com/istio/pilot
+# but we depend on being at path $GOPATH/src/istio.io/pilot for imports
+if [ "$CI" == "bootstrap" ]; then
+    mkdir -p $GOPATH/src/istio.io
+    mv $GOPATH/src/github.com/istio/pilot $GOPATH/src/istio.io
```
Comment (nlandolfi) [inline prow/pilot-presubmit.sh]: done

**Comment by nlandolfi** [comment]
adding  codecov now

**Comment by nlandolfi** [comment]
@istio-testing test this

**Comment by nlandolfi** [comment]
@istio-testing test this

**Comment by nlandolfi** [comment]
Most recent prow result [here](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio_pilot/910/pilot-presubmit/52/). I think I fixed the funkiness with zipkin by explicitly specifying an annotation query. 

@kyessenov please take another look. after that I think this is good to go.

Next steps once this is on master is to turn on reporting for Prow.

**Comment by nlandolfi** [comment]
cool @kyessenov I don't have write access but this is good to merge.

**Comment by kyessenov** [comment]
Awesome! Thank you!

**Pull request merged**